library/OrderCloud.Catalyst/DataMovement/ListAllAsync/ListAllExtensions.cs

```diff
@@ -0,0 +1,1077 @@
+﻿using Flurl.Util;
+using OrderCloud.SDK;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OrderCloud.Catalyst
+{
+	public static class ListAllExtensions
+	{
+		private const int MAX_PAGE_SIZE = 100;
+		private const string SORT = "ID";
+
+		public static async Task<List<Address>> ListAllAsync(this IAddressesResource resource, string buyerID, object filters = null, string accessToken = null)
+		{
+			return await ListAllHelper.ListAllAsync((page, filter) =>
+			{
+				return resource.ListAsync(buyerID, null, null, SORT, page, MAX_PAGE_SIZE, filters.AndFilter(filter), accessToken);
+			});
+		}
+
+		public static async Task<List<T>> ListAllAsync<T>(this IAddressesResource resource, string buyerID, object filters = null, string accessToken = null)
+			where T : Address
+		{
+			return await ListAllHelper.ListAllAsync((page, filter) =>
+			{
+				return resource.ListAsync<T>(buyerID, null, null, SORT, page, MAX_PAGE_SIZE, filters.AndFilter(filter), accessToken);
+			});
+		}
+
+		public static async Task<List<Address>> ListAllAsync(this IAdminAddressesResource resource, object filters = null, string accessToken = null)
```

Comment:
Can you add support for custom address models that extend the base class? So for example we could use HSAddress here instead of Address.

I think it would look something like this:
```csharp
public static async Task<List<TAddress>> ListAllAsync(this IAdminAddressesResource resource, object filters = null, string accessToken = null) where TAddress: Address
```

Reply:
I think that already exists. Go look at line 41.